server/src/main/java/org/elasticsearch/search/retriever/KnnRetrieverBuilder.java

```diff
@@ -261,6 +280,12 @@ public void doToXContent(XContentBuilder builder, Params params) throws IOExcept
         if (similarity != null) {
             builder.field(VECTOR_SIMILARITY.getPreferredName(), similarity);
         }
+
+        if (rescoreVectorBuilder != null) {
+            builder.startObject(RESCORE_FIELD.getPreferredName());
+            rescoreVectorBuilder.toXContent(builder, params);
+            builder.endObject();
```

Comment:
seems like this should just be `builder.field(RESCORE_FIELD.getPreferredName(), rescoreVectorBuilder)` ?